package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -1002,32 +969,52 @@ export const MessageInputProvider = <
   }) => {
     const id = generateRandomId();
     const mimeType = lookup(file.name);
+
+    const blockedFile = blockedFiles?.some((x: string) => file.name?.includes(x));
+
     const newFile = {
       file: { ...file, type: mimeType || file?.type },
       id,
       state: FileState.UPLOADING,
     };
+
     await Promise.all([
       setFileUploads((prevFileUploads) => prevFileUploads.concat([newFile])),
       setNumberOfUploads((prevNumberOfUploads) => prevNumberOfUploads + 1),
     ]);
 
-    uploadFile({ newFile });
+    if (blockedFile) {
+      newFile.state = FileState.NOT_SUPPORTED;
+    } else {
```

Comment:
Why is this state being set _after_ we set file uploads? If it's done in this order, this code does nothing in practice